### PR TITLE
GH2259: Show warning when referenced package is missing version number

### PR DESCRIPTION
**Pull request by Marusyk**
Please review. If it will be ok then I add some unit tests. #2255

Thanks in advance.

### CODE REVIEW

File: src/Cake.Core/Scripting/ScriptProcessor.cs [hunk 1]
```diff
@@ -218,5 +228,28 @@ private IPackageInstaller GetInstaller(PackageReference package, PackageType typ
                 }
             }
         }
+
+        private void CheckPackageVersion(PackageReference packageReference)
+        {
+            bool existsVersionParameter = packageReference.Parameters.Any(x => x.Key.Equals("version", StringComparison.OrdinalIgnoreCase));
+            if (!existsVersionParameter)
+            {
+                const string message = "The '{0}' directive is attempting to install the '{1}' package \r\n" +
```
[the remark this answers is not in the document]
Comment (Marusyk) [reply src/Cake.Core/Scripting/ScriptProcessor.cs]: screenshot_1

File: src/Cake.Core/Scripting/ScriptProcessor.cs [hunk 2]
```diff
@@ -189,6 +197,8 @@ private IPackageInstaller GetInstaller(PackageReference package, PackageType typ
                 _log.Verbose($"Installing {packageTypeName}s...");
                 foreach (var tool in modules)
                 {
+                    CheckPackageVersion(tool);
```
[the remark this answers is not in the document]
Comment (Marusyk) [reply src/Cake.Core/Scripting/ScriptProcessor.cs]: No, you are not right. Please see the code. The `CheckPackageVersion()` calls inside `InstallPackages()`.  `InstallModules()` and `InstallTools()` just use `InstallPackages()`. So, we have one call for Addins and one for Modules and Tools

File: src/Cake.Core/Scripting/ScriptProcessor.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (gep13) [reply src/Cake.Core/Scripting/ScriptProcessor.cs]: What you have `nuget:?package=Cake.Kudu.Client directive` I had intended to see either `tool`, `addin` or `module`.  So the message would read:

`The tool directive is attempting to install the Cake.Kudu.Client package...`

[the remark this answers is not in the document]
Comment (gep13) [reply src/Cake.Core/Scripting/ScriptProcessor.cs]: @cake-build/cake-team thoughts?

File: src/Cake.Core/Scripting/ScriptProcessor.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (gep13) [reply src/Cake.Core/Scripting/ScriptProcessor.cs]: Ah, my bad!  Sorry, I was basing it solely on the diff, which didn't show all the methods that it was actually in 😄

File: src/Cake.Core/Scripting/ScriptProcessor.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Marusyk) [reply src/Cake.Core/Scripting/ScriptProcessor.cs]: fixed

**Comment by devlead** [comment]
@Marusyk your changes have been merged, thanks for your contribution 👍

**Pull request merged**